packages/react-native/UPGRADING.md

```diff
@@ -0,0 +1,279 @@
+
+Since this update involved a major bump to Bugsnag's JS, Android and iOS notifiers – bringing all of their APIs in sync, virtually every usage of Bugsnag's API in your application will need updating.
+
+#### Initialization
+
+Previously in `bugsnag-react-native` it was possible to initialize in a variety of ways. Now there is one way to do it.
+
+In each of your native projects you need to initialize Bugsnag. This was optional before, but it means that now crashes before the JS layer loads will always be caught.
+
+To facilitate this, and to get Bugsnag started as early as possible, most configuration can only be done in the native layer. Some JS-specific options are allowed in the JS layer.
+
+##### iOS
+
+Update any of the following imports in. These will likely be in `AppDelegate.m` where it's recommended to configure Bugsnag, but may be in other locations too:
+
+```diff
+- #import <BugsnagReactNative/BugsnagReactNative.h>
+- #import "BugsnagConfiguration.h"
++ import <Bugsnag/Bugsnag.h>
+```
+
+Inside your `application:didFinishLaunchingWithOptions` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- [BugsnagReactNative start];
++ [Bugsnag start];
+```
+
+Or if you need to customize your configuration:
+
+```diff
+- BugsnagConfiguration *config = [BugsnagConfiguration new];
+- config.releaseStage = @"beta";
+- [BugsnagReactNative startWithConfiguration:config];
++ BugsnagConfiguration *config = [BugsnagConfiguration loadConfig];
++ config.releaseStage = @"beta";
++ [Bugsnag startWithConfiguration:config];
+```
+
+See the new [Bugsnag Cocoa configuration guide](https://docs.bugsnag.com/platforms/ios/configuration-options/) for all available configuration options.
+
+If you have set your API key in `Info.plist`, the format has changed from a top level string to a dictionary.
+
+```
+- <key>BugsnagAPIKey</key>
+- <string>YOUR_API_KEY</string>
++ <key>bugsnag</key>
++ <dict>
++     <key>apiKey</key>
++     <string>YOUR_API_KEY</string>
++ </dict>
+```
+
+##### Android
+
+Replace the following import which will likely be in your `MainApplication.java` file:
+
+```diff
+- import com.bugsnag.android.BugsnagReactNative;
++ import com.bugsnag.android.Bugsnag;
+```
+
+Inside your application's `onCreate` method, replace any Bugsnag initialization code with the updated version:
+
+```diff
+- Configuration config = new Configuration();
+- config.setReleaseStage("beta");
+- BugsnagReactNative.startWithConfiguration(this /* app context */, config);
++ Configuration config = Configuration.load(this);
++ config.setReleaseStage("beta");
+Bugsnag.start(this /* app context */, config);
+```
+
+###### JS
+
+The JS interface has been updated to match our new notifier spec standard – that means it looks the same as our JS notifier, and uses the same names and concepts as our other platforms.
```

Comment:
Do we want to refer to the spec?

```suggestion
As part of the `bugsnag-js monorepo`, the JS interface now looks the same as our JS notifier, and uses the same names and concepts as our other platforms.
```

Reply:
I figured it was ok to refer to in passing, but happy with your suggestion.